Libraries/xcassets/Headers/xcassets/Slot/ImageSize.h

```diff
@@ -0,0 +1,52 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#ifndef __xcassets_Slot_ImageSize_h
+#define __xcassets_Slot_ImageSize_h
+
+#include <ext/optional>
+#include <vector>
+#include <string>
+
+namespace xcassets {
+namespace Slot {
+
+/*
+ * The density ratio of a target device.
+ */
+class ImageSize {
+private:
+    std::vector<double> _value;
+
+private:
+    ImageSize(std::vector<double> imageSize);
```

Comment:
I'll switch to width/height. The runtime warnings were bugging me so I just did a quick implementation.